src/Hpack/Config.hs

```diff
@@ -212,6 +214,20 @@ instance HasFieldNames LibrarySection
 instance FromJSON LibrarySection where
   parseJSON = genericParseJSON_
 
+data InternalLibrarySection = InternalLibrarySection {
+  internalLibrarySectionExposedModules :: Maybe (List String)
+, internalLibrarySectionOtherModules :: Maybe (List String)
+, internalLibrarySectionReexportedModules :: Maybe (List String)
+} deriving (Eq, Show, Generic)
```

Comment:
The only difference between `InternalLibrarySection` and `LibrarySection` is that `InternalLibrarySection` won't accept `exposed`.  However, from what I tried, `Cabal` will accept `exposed` for internal libraries.  I'm not sure what exactly the effect is (if any).  But for code simplicity, let's follow what cabal does and use `LibrarySection` instead of `InternalLibrarySection`.